snforge_std/src/cheatcodes/l1_handler.cairo

```diff
@@ -5,29 +5,36 @@ use super::super::_cheatcode::handle_cheatcode;
 
 #[derive(Drop, Clone)]
 struct L1Handler {
-    contract_address: ContractAddress,
-    function_selector: felt252,
-    from_address: felt252,
-    payload: Span::<felt252>,
+    target: ContractAddress,
+    selector: felt252,
 }
 
 trait L1HandlerTrait {
-    fn new(contract_address: ContractAddress, function_selector: felt252) -> L1Handler;
-    fn execute(self: L1Handler) -> SyscallResult<()>;
+    fn new(target: ContractAddress, selector: felt252) -> L1Handler;
+    fn execute(
+        self: L1Handler, from_address: felt252, payload: Span::<felt252>
+    ) -> SyscallResult<()>;
 }
 
 impl L1HandlerImpl of L1HandlerTrait {
-    fn new(contract_address: ContractAddress, function_selector: felt252) -> L1Handler {
-        L1Handler {
-            contract_address, function_selector, from_address: 0, payload: array![].span(),
-        }
+    /// `target` - The target starknet contract address
+    /// `selector` - Selector of a `#[l1_handler]` function
+    /// Returns a structure referring to a L1 handler function
+    fn new(target: ContractAddress, selector: felt252) -> L1Handler {
+        L1Handler { target, selector, }
     }
 
-    fn execute(self: L1Handler) -> SyscallResult<()> {
+    /// `self` - `L1Handler` structure referring to a L1 handler function
+    /// `from_address` - Ethereum address of the contract that you want to a emulate message from
```

Comment:
```suggestion
    /// `from_address` - Ethereum address of the contract that you want to be the message sender
```

Reply:
I think my description better describes the reality, It clearly states that it is emulated.

Reply:
It's not the job of the comment about the parameter to explain that the message is emulated though

Reply:
Good point, I will adjust it